modules/i18n/i18n.lua

```diff
@@ -5,17 +5,45 @@ local i18n = VFS.Include(I18N_PATH .. "init.lua", nil, VFS.ZIP)
 local asianFont = 'SourceHanSans-Regular.ttc'
 local translationDirs = VFS.SubDirs('language')
 
+-- Construct a map of
+-- languageCode -> list of translation files associated with that language.
+local languageFiles = {}
 for _, languageDir in ipairs(translationDirs) do
 	local translationFiles = VFS.DirList(languageDir, '*.json')
 	local languageCode = table.remove( string.split(languageDir, '/') )
+	languageFiles[languageCode] = translationFiles
+end
+
+-- Loads all language translation files associated with languageCode.
+local function loadLanguageFiles(languageCode)
+	Spring.Log("i18n", "debug", "Loading " .. languageCode .. " translation files.")
+	if languageFiles[languageCode] == nil then
+		Spring.Log("i18n", "warning",
+				"No language files associated with requested language code " ..
```

Comment:
Why is this logging needed? We are 100% in control of these files.

Reply:
Removed the log lines.